### PR TITLE
feat: New open source models

### LINKED ISSUE #648: [Feature Request] More open source models support
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

Recently many new open soure models has been released, we can add to them to camel.

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zhiyu-01**
## Description

Add llama-3, qwen-2 glm-4 to camel.

## Motivation and Context

close #648 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/types/enums.py [hunk 1]
```diff
@@ -145,6 +154,12 @@ def token_limit(self) -> int:
             return 4096
         elif self is ModelType.LLAMA_2:
             return 4096
+        elif self is ModelType.LLAMA_3:
+            return 8192
+        elif self is ModelType.QWEN_2:
+            return 128000
+        elif self is ModelType.GLM_4:
+            return 8192
         elif self is ModelType.VICUNA:
```
Comment (Wendong-Fan) [inline camel/types/enums.py:163]: duplicated
```suggestion
            return 128000
        elif self is ModelType.VICUNA:
```

File: camel/utils/token_counting.py [hunk 2]
```diff
@@ -179,8 +179,23 @@ def count_tokens_from_messages(self, messages: List[OpenAIMessage]) -> int:
...
+                messages,
+                add_generation_prompt=True,
+                tokenize=True,
+                return_tensors="pt",
+                return_dict=True,
+            )
+            input_ids = inputs["input_ids"]
+        else:
+            prompt = messages_to_prompt(messages, self.model_type)
+            input_ids = self.tokenizer(prompt).input_ids
```
Comment (Wendong-Fan) [inline camel/utils/token_counting.py:198]: Hey @zhiyu-01 ,could you help me understand why the set for token counter here is different for `QWEN_2` and `GLM_4_OPEN_SOURCE`? Any reference for this? Thanks in advance!

**Review by Wendong-Fan — COMMENTED** [review]
Thanks @zhiyu-01 ! Overall it's great, just one question in below~

File: camel/utils/token_counting.py (hunk 2 above)
Comment (zhiyu-01) [reply camel/utils/token_counting.py:198]: https://huggingface.co/THUDM/glm-4-9b-chat
https://huggingface.co/Qwen/Qwen2-7B-Instruct

I found these two settings from their huggingface repo.

File: camel/types/enums.py (hunk 1 above)
Comment (zhiyu-01) [reply camel/types/enums.py:163]: Changed.

File: camel/utils/token_counting.py (hunk 2 above)
Comment (Wendong-Fan) [reply camel/utils/token_counting.py:198]: Thanks for the answer, actually I think the 2 code blocks are doing the same thing, maybe we can unify the code

Comment (zhiyu-01) [reply camel/utils/token_counting.py]: I have run the models, and set the str prompt correspondingly.
qwen:
qwen
glm:
glm

**Review by Wendong-Fan — APPROVED** [review]

**Pull request merged**